File: Bridge/System/Threading/ManualResetEventSlim.cs
using Bridge;

namespace System.Threading
{
    public class ManualResetEventSlim : IDisposable
    {
        public ManualResetEventSlim()
            : this(false, 10)
        {
        }

        public ManualResetEventSlim(bool initialState)
            : this(initialState, 10)
        {
        }

        public ManualResetEventSlim(bool initialState, int spinCount)
        {
        }

        public bool IsSet
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public int SpinCount
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public void Reset()
        {
            throw new NotImplementedException();
        }

        public void Set()
        {
            throw new NotImplementedException();
        }
        
        public void Wait()
        {
            throw new NotImplementedException();
        }

        public bool Wait(int millisecondsTimeout)
        {
            throw new NotImplementedException();
        }

        public bool Wait(TimeSpan timeout)
        {
            throw new NotImplementedException();
        }

        public void Wait(CancellationToken cancellationToken)
        {
            throw new NotImplementedException();
        }

        public bool Wait(int millisecondsTimeout, CancellationToken cancellationToken)
        {
            throw new NotImplementedException();
        }

        public bool Wait(TimeSpan timeout, CancellationToken cancellationToken)
        {
            throw new NotImplementedException();
        }

        public WaitHandle WaitHandle
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public void Dispose()
        {
            throw new NotImplementedException();
        }

        protected virtual void Dispose(bool disposing)
        {
            throw new NotImplementedException();
        }
        
        static int CheckTimeout(TimeSpan timeout)
        {
            throw new NotImplementedException();
        }
    }
}

File: Bridge/Resources/.generated/system/threading/manualResetEventSlim.js
    Bridge.define("System.Threading.ManualResetEventSlim", {
        inherits: [System.IDisposable],
        statics: {
            methods: {
                checkTimeout: function (timeout) {
                    throw new System.NotImplementedException();
                }
            }
        },
        props: {
            IsSet: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            SpinCount: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            WaitHandle: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            }
        },
        alias: ["dispose", "System$IDisposable$dispose"],
        ctors: {
            ctor: function () {
                System.Threading.ManualResetEventSlim.$ctor2.call(this, false, 10);
            },
            $ctor1: function (initialState) {
                System.Threading.ManualResetEventSlim.$ctor2.call(this, initialState, 10);
            },
            $ctor2: function (initialState, spinCount) {
                this.$initialize();
            }
        },
        methods: {
            reset: function () {
                throw new System.NotImplementedException();
            },
            set: function () {
                throw new System.NotImplementedException();
            },
            wait$4: function () {
                throw new System.NotImplementedException();
            },
            wait: function (millisecondsTimeout) {
                throw new System.NotImplementedException();
            },
            wait$2: function (timeout) {
                throw new System.NotImplementedException();
            },
            wait$5: function (cancellationToken) {
                throw new System.NotImplementedException();
            },
            wait$1: function (millisecondsTimeout, cancellationToken) {
                throw new System.NotImplementedException();
            },
            wait$3: function (timeout, cancellationToken) {
                throw new System.NotImplementedException();
            },
            dispose: function () {
                throw new System.NotImplementedException();
            },
            dispose$1: function (disposing) {
                throw new System.NotImplementedException();
            }
        }
    });
